src/tiled/invertyaxishelper.h

```diff
@@ -24,32 +24,41 @@
 #include "preferences.h"
 #include "mapdocumentactionhandler.h"
 
-class InvertYAxisHelper{
-    public:
-        InvertYAxisHelper() {}
-        ~InvertYAxisHelper() {}
-        
-        // Inverts Y coordinate in grid
-        float getY(float y) const
+namespace Tiled{
+    namespace Internal {
+
+        class InvertYAxisHelper
         {
-            // Check if Invert Y Axis is set
-            if(Tiled::Internal::Preferences::instance()->invertYAxis())
+        public:
+            // Constructors
+            InvertYAxisHelper()  {}
+            InvertYAxisHelper(Map* m) : target(m) {}
```

Comment:
You'll need to take a `MapDocument` and not a bare `Map` (see your implementation of `tileY`, which still uses the `MapDocumentActionHandler`).

Coding style: please put the `*` against the variable name, so `MapDocument *target`.

Reply:
Thanks for the feedback, but i would like to understand the logic behind taking the `MapDocument `and not the `Map`. From my point of view, you are only going to use the Map, so what's the reason behind giving the whole `MapDocument` to the helper class?

Anyway I'll make the changes ASAP!

Reply:
Because just the map is not enough to know where the origin is for implementing `pixelY`. It is currently hardcoded to work for orthogonal maps, but it is returning the wrong thing for staggered or isometric maps. See https://github.com/bjorn/tiled/pull/1897/files#r174102431